external_search.py

```diff
@@ -17,12 +17,25 @@ class ExternalSearchIndex(object):
     work_document_type = 'work-type'
     __client = None
 
+    CURRENT_ALIAS_SUFFIX = '-current'
+    VERSION_RE = re.compile('-v([0-9]+)$')
+
+    @classmethod
+    def reset(cls):
+        """Resets the __client object to None so a new configuration
+        can be applied during object initialization.
+
+        This method is only intended for use in testing.
+        """
+        cls.__client = None
+
     def __init__(self, url=None, works_index=None):
     
         self.log = logging.getLogger("External search index")
 
         # By default, assume that there is no search index.
         self.works_index = None
+        from_configuration = False
```

Comment:
Makes total sense, and that's almost what's happening right now: we create the alias or the index as needed. (I think the code flow is confusing. I'll add some more docstrings.)

The _only_ difference is that if you provide `works-v2` in configuration, and the `works-current` alias is set on `some-other-index` then the alias is moved from `some-other-index` to `works-v2`. Sounds like this isn't what we want. My assumption was that if you've changed to a different index name in config, that's the index you want to be using in production and thus should be "current".

So in that case, should the expectation be that you'll be point to `works-v2` without the alias or should the expectation be that you'll be pointing to `some-other-index` and the `works-current` alias despite what's in your configuration?